### DISCUSSION TITLE
Understanding multi-process queue

### DISCUSSION THREAD

**Discussion by maybephilipp** [miguelgrinberg/python-socketio#1153] [Q&A] [answered]
Hello! Could someone please explain me how does multi-process queue work with different message queues in case of emitting an event to a room?

As I see in the code, participants of rooms are stored locally in each particular process, so the particular process knows only room subscribers that are connected only to this process.

I'm new into MQs so sorry if it's a stupid question.

As I understood:

- RabbitMQ sends messages in round robin model. So each message is received only by one consumer (process);
- Redis pub/sub sends messages to all consumers. So each message is received by each consumer (process);
- SQS FIFO sends messages in something like round robin. So each message is received only by one consumer (process);

I see that you support all of these MQs (SQS via kombu), but I don't understand how you accomplish consistency in all of them.

So, as I understand, in case of RabbitMQ and SQS, message will be received by only one process of my app and that means that only small part of the room get the emitted message, right?

And only in case of Redis-like emits to room will reach all room participants?

Isn't it a bug?

Thank you,
Philipp

**Comment by miguelgrinberg** [comment, marked as answer]
See the [fanout exchange](https://www.rabbitmq.com/tutorials/amqp-concepts.html#exchange-fanout).

**Reply by maybephilipp** [reply]
Sadly, Kombu doesn't support SQS + SNS Fanout. Thank you for reply.
